### ISSUE TITLE
enable end user to specify different subnets for different agentpools if they want to

### ISSUE THREAD

**Issue by jim-minter** [openshift/openshift-azure#537]
after #533 merges and we've talked about it, we should probably ensure that all cluster agent pool subnets are disjoint.  Didn't want to do this now in case it uncovered a bug at the last minute.

**Comment by mjudeikis** [comment]
@jim-minter if I understand this right we want to make sure we have something like this:
```
masterPoolProfile:
    subnetCidr: 10.0.0.0/24
  agentPoolProfiles:
  - name: infra
    subnetCidr: 10.0.0.1/24
  - name: compute
    subnetCidr: 10.0.0.2/24
```
and not:
```
masterPoolProfile:
    subnetCidr: 10.0.0.0/24
  agentPoolProfiles:
  - name: infra
    subnetCidr: 10.0.0.0/24
  - name: compute
    subnetCidr: 10.0.0.1/24
```

and at the same time all these 3 should be part of `vnetCidr` and not part of Services and pods networks?

**Comment by mjudeikis** [comment]
@jim-minter can you confirm my thinking for this?

**Comment by mjudeikis** [comment]
We should do:
1. check if subnets are unique
2. Check if they are disjoint
3. check if they are within cidr of vnet
4. Check if they are not part of services and pods subnets

naming of subnets: 
10.0.0.0/24 10-0-0-0-24

**Comment by kargakis** [comment]
Closing in favor of https://jira.coreos.com/browse/AZURE-215
